src/cli/bin.js

```diff
@@ -3,79 +3,65 @@
 /* eslint-disable no-console */
 'use strict'
 
-process.on('uncaughtException', (err) => {
-  console.info(err)
-
-  throw err
-})
-
-process.on('unhandledRejection', (err) => {
-  console.info(err)
-
-  throw err
-})
-
 const semver = require('semver')
-const pkg = require('../../package.json')
-
-if (!semver.satisfies(process.versions.node, pkg.engines.node)) {
-  console.error(`Please update your Node.js version to ${pkg.engines.node}`)
-  process.exit(1)
-}
-
 const YargsPromise = require('yargs-promise')
 const updateNotifier = require('update-notifier')
-const { print } = require('./utils')
-const mfs = require('ipfs-mfs/cli')
 const debug = require('debug')('ipfs:cli')
 const parser = require('./parser')
 const commandAlias = require('./command-alias')
+const { print } = require('./utils')
+const pkg = require('../../package.json')
 
-function main (args) {
-  const oneWeek = 1000 * 60 * 60 * 24 * 7
-  updateNotifier({ pkg, updateCheckInterval: oneWeek }).notify()
+// Handle any uncaught error
+process.once('uncaughtException', (err, origin) => {
+  if (origin === 'uncaughtException') {
+    print(err.message)
+    debug(err)
+  }
+})
+process.once('unhandledRejection', (err) => {
+  print(err.message)
+  debug(err)
+})
```

Comment:
Can we log these out as they were before? ...(maybe `console.error` not `console.info`?)

These unexpected errors aren't always easy to reproduce so getting hold of a stack trace after the fact could be difficult and time consuming. You'd have to run the command again in debug mode and somehow replicate the circumstances that trigger the error.

Users also rarely run in debug mode so won't see the stack either. That translates to it being more difficult for us to begin to debug if users open an issue with just an error message and not a stack trace.

It also makes it harder for users to self debug - users unfamiliar with the `debug` module need to figure out how to turn it on in order to get the stack.

Reply:
to do that it's better to just let them throw and remove the handlers all together

Reply:
<img width="468" alt="Screenshot 2019-07-23 at 14 00 14" src="https://user-images.githubusercontent.com/152863/61714248-49e24580-ad52-11e9-82fa-c798023a4795.png">

First is with:

```js
process.on('uncaughtException', err => {
  console.error(err)
  process.exit(1)
})
```

...second is without.

Not much in it TBH. It's kinda nice to have the Node.js internal stack greyed out.

My main concern is having the stack logged for these unexpected errors and not behind a debug flag 😉